spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2845,10 +2845,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           }
         }
 
-        // TODO: Support SortMergeJoin with join condition after new DataFusion release
-        if (join.condition.isDefined) {
-          withInfo(op, "Sort merge join with a join condition is not supported")
-          return None
+        val condition = join.condition.map { cond =>
+          val condProto = exprToProto(cond, join.left.output ++ join.right.output)
+          if (condProto.isEmpty) {
+            withInfo(join, cond)
+            return None
+          }
+          condProto.get
```

Comment:
is there any scenario of `None.get`?

Reply:
```scala
if (condProto.isEmpty) {
  withInfo(join, cond)
   return None
}
```

If it is `None`, it will return `None` earlier.